app/js/directives/CentreDirective.js

```diff
@@ -9,17 +9,16 @@ function CentreDirective() {
     },
     link: function (scope, element, attrs) {
       scope.$watch('attributes.maleAvailability', function (newvalue) {
-        scope.attributes.highlightMale = newvalue < 1 ? 'highlight' : '';
-        scope.attributes.maleAvailability = newvalue === 0 ? 'FULL' : newvalue;
-        scope.attributes.statusMale = newvalue < 1 ? 'No Available Male Beds' : 'Available Male Beds';
-
-      });
```

Comment:
$watch(watchExpression, listener, [objectEquality]);

1. **objectEquality = false** :: Inequality is determined according to reference inequality, strict comparison via the !== Javascript operator, 

2. **objectEquality = true** :: inequality of the watchExpression is determined according to the angular.equals function. angular.equals (relevant for complex objects not integers) : 

- Both objects or values pass === comparison.

**'attributes.maleAvailability' relevant (and the equivalent of objectEquality = false)**

- Both objects or values are of the same type and all of their properties are equal by comparing them with angular.equals. 

**'attributes.maleAvailability' is not an object.** 

- Both values are NaN. (In JavaScript, NaN == NaN => false. But we consider two NaN as equal)

**'attributes.maleAvailability' will not be NaN**

- Both values represent the same regular expression (In JavaScript, /abc/ == /abc/ => false. But we consider two regular expressions as equal when their textual representation matches).

**'attributes.maleAvailability' is not a regular expression.**